jujugui/static/gui/src/app/store/env/base.js

```diff
@@ -275,10 +275,13 @@ YUI.add('juju-env-base', function(Y) {
 
       @method close
     */
-    close: function() {
+    close: function(cb) {
       if (this.ws) {
         this.beforeClose(this.ws.close.bind(this.ws));
       }
+      if (cb) {
+        cb();
```

Comment:
env.on_close and env.connect appear to cause side effects we need. I can't quite suss out what b/c our env code is pretty complex, but empirically this worked, and dropping them did not.